### PR TITLE
improve modules doc generation by support macros

### PR DESCRIPTION
**Pull request by thiolliere**
related to https://github.com/paritytech/substrate/issues/1917

It is not fully defined definitive doc but at least it makes a start for future improvment which will rely on modifying doc only.

Also I'm still mixed about something: Documentation passed in decl_module appears for Call enum which makes sense, but then Module structure has no documentation. it may needs none, I don't know what to write on it.

Now except Module struct all type of a module can be documented with macro or have default documentation.

### CODE REVIEW

File: srml/support/src/event.rs [hunk 1]
```diff
@@ -227,6 +227,9 @@ macro_rules! __decl_generic_event {
 		{ $( $events:tt )* };
 		{ ,$( $generic_param:ident = $generic_type:ty ),* };
 	) => {
+		/// Accessor to [`RawEvent`]
```
Comment (bkchr) [inline srml/support/src/event.rs]: Hmm, I'm not quite happy with this explanation.

File: srml/support/src/dispatch.rs [hunk 2]
```diff
@@ -601,18 +593,25 @@ macro_rules! decl_module {
...
-				$mod_type<$trait_instance: $trait_name>;
-				$origin_type;
-				$from;
-				$(#[doc = $doc_attr])*
-				$fn_vis fn $fn_name (
-					$from $(, $param_name : $param )*
-				) $( -> $result )* { $( $impl )* }
-			}
-		)*
+		/// Public dispatchable functions.
```
Comment (bkchr) [inline srml/support/src/dispatch.rs]: Is correct, but all dispatchable functions/calls are "public". The ones that will be visible in the documentation are somewhat special, as being public functions.

File: srml/support/src/event.rs (hunk 1 above)
Comment (joepetrowski) [reply srml/support/src/event.rs]: `RawEvent` specialized for the configuration `Trait`

File: srml/support/src/dispatch.rs (hunk 2 above)
Comment (thiolliere) [reply srml/support/src/dispatch.rs]: should I write "exposed dispatchable functions" here ? or just remove the line

File: srml/support/src/event.rs (hunk 1 above)
Comment (thiolliere) [reply srml/support/src/event.rs]: I have written as such

File: srml/support/src/dispatch.rs (hunk 2 above)
Comment (thiolliere) [reply srml/support/src/dispatch.rs]: I removed the line

**Comment by bkchr** [comment]
Please merge master.

**Pull request merged**